src/mpc-hc/SubtitleUpDlg.cpp

```diff
@@ -0,0 +1,517 @@
+        m.AppendMenu(MF_STRING | MF_ENABLED, OPEN_URL, ResStr(IDS_SUBMENU_OPENURL));
+
+        CPoint pt = lpnmlv->ptAction;
+        ::MapWindowPoints(lpnmlv->hdr.hwndFrom, HWND_DESKTOP, &pt, 1);
+
+        switch (m.TrackPopupMenu(TPM_LEFTBUTTON | TPM_RETURNCMD, pt.x, pt.y, this)) {
+            case OPEN_URL:
+                provider.OpenUrl();
+                break;
+            case SET_CREDENTIALS: {
+                CString szUser(UTF8To16(provider.UserName().c_str()));
+                CString szPass(UTF8To16(provider.Password().c_str()));
+                CString szDomain(provider.Name().c_str());
+                if (ERROR_SUCCESS == PromptForCredentials(GetSafeHwnd(),
+                                                          ResStr(IDS_SUB_CREDENTIALS_TITLE), ResStr(IDS_SUB_CREDENTIALS_MSG) + CString(provider.Url().c_str()),
+                                                          szDomain, szUser, szPass, /*&bSave*/nullptr)) {
+                    provider.UserName((const char*)UTF16To8(szUser));
+                    provider.Password((const char*)UTF16To8(szPass));
+                    m_list.SetItemText(lpnmlv->iItem, 1, szUser);
+                    s.strSubtitlesProviders = CString(m_MainFrame.m_pSubtitlesProviders->WriteSettings().c_str());
+                    s.SaveSettings();
+                }
+                break;
+            }
+            case RESET_CREDENTIALS:
+                provider.UserName("");
+                provider.Password("");
+                m_list.SetItemText(lpnmlv->iItem, 1, _T(""));
+                s.strSubtitlesProviders = CString(m_MainFrame.m_pSubtitlesProviders->WriteSettings().c_str());
+                s.SaveSettings();
+                break;
+            case MOVE_UP: {
+                m_MainFrame.m_pSubtitlesProviders->MoveUp(lpnmlv->iItem);
+                ListView_SortItemsEx(m_list.GetSafeHwnd(), SortCompare, m_list.GetSafeHwnd());
+                s.strSubtitlesProviders = CString(m_MainFrame.m_pSubtitlesProviders->WriteSettings().c_str());
+                s.SaveSettings();
+                break;
+            }
+            case MOVE_DOWN: {
+                m_MainFrame.m_pSubtitlesProviders->MoveDown(lpnmlv->iItem);
+                ListView_SortItemsEx(m_list.GetSafeHwnd(), SortCompare, m_list.GetSafeHwnd());
+                s.strSubtitlesProviders = CString(m_MainFrame.m_pSubtitlesProviders->WriteSettings().c_str());
+                s.SaveSettings();
+                break;
+            }
+            default:
+                break;
+        }
+    }
+}
+
+int CALLBACK CSubtitleUpDlg::SortCompare(LPARAM lParam1, LPARAM lParam2, LPARAM lParamSort)
+{
+    CListCtrl& list = *(CListCtrl*)CListCtrl::FromHandle((HWND)(lParamSort));
+    size_t left = ((SubtitlesProvider*)list.GetItemData((int)lParam1))->Index();
+    size_t right = ((SubtitlesProvider*)list.GetItemData((int)lParam2))->Index();
+    return left == right ? 0 : (int)(left - right);
```

Comment:
I will look at the details on Friday but since I spotted those:
`(HWND)(lParamSort)` --> `(HWND)lParamSort`
`left == right ? 0 : (int)(left - right)` --> `(int)(left - right)` or even `int(left - right)`